dojo/unittests/test_nexpose_parser.py

```diff
@@ -0,0 +1,16 @@
+from django.test import TestCase
+from dojo.tools.nexpose.parser import NexposeFullXmlParser
+from dojo.models import Test
+
+
+class TestNexposeParser(TestCase):
+    def test_nexpose_parser_has_no_finding(self):
+        testfile = open("dojo/unittests/scans/nexpose/no_vuln.xml")
+        parser = NexposeFullXmlParser(testfile, Test())
+        self.assertEqual(0, len(parser.items))
+
+    def test_nexpose_parser_has_many_finding(self):
+        testfile = open("dojo/unittests/scans/nexpose/many_vulns.xml")
+        parser = NexposeFullXmlParser(testfile, Test())
+        testfile.close()
+        self.assertEqual(125, len(parser.items))
```

Comment:
could you add test on `unsaved_endpoints` attribute?

Reply:
I will test it on my docker and tell the result asap

Reply:
This error log 
Run docker-compose up --exit-code-from integration-tests integration-tests
using --exit-code-from implies --abort-on-container-exit
django-defectdojo_mysql_1 is up-to-date
django-defectdojo_uwsgi_1 is up-to-date
django-defectdojo_nginx_1 is up-to-date
Creating django-defectdojo_integration-tests_1 ... 

Creating django-defectdojo_integration-tests_1 ... done
Attaching to django-defectdojo_integration-tests_1
integration-tests_1  | wait-for-it.sh: waiting max 30 seconds for mysql:3306
integration-tests_1  | wait-for-it.sh: mysql:3306 is available after 0 seconds
integration-tests_1  | Testing DefectDojo Service
integration-tests_1  | Waiting max 60s for services to start
integration-tests_1  | Succesfully displayed login page, starting integration tests
integration-tests_1  | Running: Report Builder tests
integration-tests_1  | test_login (base_test_class.BaseTestCase) ... ok
integration-tests_1  | test_create_product (product_test.ProductTest) ... ok
integration-tests_1  | test_add_product_finding (product_test.ProductTest) ... ok
integration-tests_1  | test_add_product_endpoints (product_test.ProductTest) ... ok
integration-tests_1  | generate_HTML_report (__main__.ReportBuilderTest) ... ok
integration-tests_1  | generate_AsciiDoc_report (__main__.ReportBuilderTest) ... launching browser for:  BaseTestCase
integration-tests_1  | starting chromedriver with options:  {'_binary_location': '', '_arguments': ['--headless', '--no-sandbox', '--disable-gpu', '--window-size=1280,1024'], '_extension_files': [], '_extensions': [], '_experimental_options': {}, '_debugger_address': None, '_caps': {'browserName': 'chrome', 'version': '', 'platform': 'ANY', 'acceptInsecureCerts': True}} {'browserName': 'chrome', 'version': '', 'platform': 'ANY', 'goog:loggingPrefs': {'browser': 'ALL'}}
integration-tests_1  | ×
integration-tests_1  | Product added successfully.
integration-tests_1  | ×
integration-tests_1  | Endpoint added successfully.
integration-tests_1  | {'level': 'SEVERE', 'message': "http://nginx:8080/static/bootstrap/dist/js/bootstrap.min.js 5:26266 Uncaught TypeError: Cannot read property 'trigger' of null", 'source': 'javascript', 'timestamp': 1611931657063}
integration-tests_1  | There was a SEVERE javascript error in the console, please check all steps fromt the current test to see where it happens
integration-tests_1  | Currently there is no reliable way to find out at which url the error happened, but it could be: .http://nginx:8080/reports/custom
integration-tests_1  | WARNING: skipping javascript errors related to finding images, see https://github.com/DefectDojo/django-DefectDojo/issues/2045
integration-tests_1  | {'level': 'SEVERE', 'message': "http://nginx:8080/static/bootstrap/dist/js/bootstrap.min.js 5:26266 Uncaught TypeError: Cannot read property 'trigger' of null", 'source': 'javascript', 'timestamp': 1611931657609}
integration-tests_1  | There was a SEVERE javascript error in the console, please check all steps fromt the current test to see where it happens
integration-tests_1  | Currently there is no reliable way to find out at which url the error happened, but it could be: .http://nginx:8080/reports/custom
integration-tests_1  | WARNING: skipping javascript errors related to finding images, see https://github.com/DefectDojo/django-DefectDojo/issues/2045
integration-tests_1  | {'level': 'SEVERE', 'message': "http://nginx:8080/static/bootstrap/dist/js/bootstrap.min.js 5:26266 Uncaught TypeError: Cannot read property 'trigger' of null", 'source': 'javascript', 'timestamp': 1611931658159}
integration-tests_1  | There was a SEVERE javascript error in the console, please check all steps fromt the current test to see where it happens
integration-tests_1  | Currently there is no reliable way to find out at which url the error happened, but it could be: .http://nginx:8080/reports/custom
integration-tests_1  | WARNING: skipping javascript errors related to finding images, see https://github.com/DefectDojo/django-DefectDojo/issues/2045
integration-tests_1  | {'level': 'SEVERE', 'message': "http://nginx:8080/static/bootstrap/dist/js/bootstrap.min.js 5:26266 Uncaught TypeError: Cannot read property 'trigger' of null", 'source': 'javascript', 'timestamp': 1611931658731}
integration-tests_1  | There was a SEVERE javascript error in the console, please check all steps fromt the current test to see where it happens
integration-tests_1  | Currently there is no reliable way to find out at which url the error happened, but it could be: .http://nginx:8080/reports/custom
integration-tests_1  | WARNING: skipping javascript errors related to finding images, see https://github.com/DefectDojo/django-DefectDojo/issues/2045
integration-tests_1  | {'level': 'SEVERE', 'message': "http://nginx:8080/static/bootstrap/dist/js/bootstrap.min.js 5:26266 Uncaught TypeError: Cannot read property 'trigger' of null", 'source': 'javascript', 'timestamp': 1611931659316}
integration-tests_1  | There was a SEVERE javascript error in the console, please check all steps fromt the current test to see where it happens
integration-tests_1  | Currently there is no reliable way to find out at which url the error happened, but it could be: .http://nginx:8080/reports/custom
integration-tests_1  | WARNING: skipping javascript errors related to finding images, see https://github.com/DefectDojo/django-DefectDojo/issues/2045
integration-tests_1  | {'level': 'SEVERE', 'message': "http://nginx:8080/static/bootstrap/dist/js/bootstrap.min.js 5:26266 Uncaught TypeError: Cannot read property 'trigger' of null", 'source': 'javascript', 'timestamp': 1611931659902}
integration-tests_1  | There was a SEVERE javascript error in the console, please check all steps fromt the current test to see where it happens
integration-tests_1  | Currently there is no reliable way to find out at which url the error happened, but it could be: .http://nginx:8080/reports/custom
integration-tests_1  | WARNING: skipping javascript errors related to finding images, see https://github.com/DefectDojo/django-DefectDojo/issues/2045
integration-tests_1  | {'level': 'SEVERE', 'message': 'http://nginx:8080/static/datatables.net-buttons/js/buttons.colVis.min.js - Failed to load resource: net::ERR_INSUFFICIENT_RESOURCES', 'source': 'network', 'timestamp': 1611931661329}
integration-tests_1  | There was a SEVERE javascript error in the console, please check all steps fromt the current test to see where it happens
integration-tests_1  | Currently there is no reliable way to find out at which url the error happened, but it could be: .http://nginx:8080/reports/custom
integration-tests_1  | tearDownDriver:  BaseTestCase
integration-tests_1  | closing browser
integration-tests_1  | FAIL
integration-tests_1  | 
integration-tests_1  | ======================================================================
integration-tests_1  | FAIL: generate_AsciiDoc_report (__main__.ReportBuilderTest)
integration-tests_1  | ----------------------------------------------------------------------
integration-tests_1  | Traceback (most recent call last):
integration-tests_1  |   File "/app/tests/base_test_class.py", line 312, in tearDown
integration-tests_1  |     self.assertNoConsoleErrors()
integration-tests_1  |   File "/app/tests/base_test_class.py", line 307, in assertNoConsoleErrors
integration-tests_1  |     self.assertNotEqual(entry['level'], 'SEVERE')
integration-tests_1  | AssertionError: 'SEVERE' == 'SEVERE'
integration-tests_1  | 
integration-tests_1  | ----------------------------------------------------------------------
integration-tests_1  | Ran 6 tests in 29.425s
integration-tests_1  | 
integration-tests_1  | FAILED (failures=1)
integration-tests_1  | Error: Report test failed\n
django-defectdojo_integration-tests_1 exited with code 1
Aborting on container exit...

Error: Process completed with exit code 1.

Reply:
I don't think it' linked to your modifications. I'm able to reproduce it in dev. We are working on fixing this.

Reply:
I am so sorry @damiencarol, I do not know much about this error. Thanks for your support!